src/common/utils.cc

```diff
@@ -47,6 +47,27 @@ QString unescapeAmps( QString const & str )
   result.replace( "&&", "&" );
   return result;
 }
+
+QTextCodec::Encoding detectCodec( QByteArray & ba, char16_t expectedFirstCharacter = 0 )
```

Comment:
This is the wrong direction. `QtCore5Compat/QTextCodec` should be removed instead of developing anything new with it.